vars/buildDaosJob.groovy

```diff
@@ -0,0 +1,32 @@
+// vars/buildDaosJob.groovy
+
+  /**
+   * buildDaosJob step method.
+   *
+   * This is intended only for use in a Jenkinsfile in a Matrix stage.
+   * Builds a branch of DAOS based on env.TEST_BRANCH.
```

Comment:
We could eliminate this requirement and the external assumption by making the branch to build an argument.

Reply:
Changed in current push